sdk/dapp-kit/src/hooks/wallet/useUnsafeBurnerWallet.ts

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Ed25519Keypair } from '@mysten/sui.js/keypairs/ed25519';
+import { SuiClient, getFullnodeUrl } from '@mysten/sui.js/client';
+import type {
+	StandardConnectFeature,
+	StandardConnectMethod,
+	StandardEventsFeature,
+	StandardEventsOnMethod,
+	SuiFeatures,
+	SuiSignPersonalMessageMethod,
+	SuiSignTransactionBlockMethod,
+	SuiSignAndExecuteTransactionBlockMethod,
+	Wallet,
+} from '@mysten/wallet-standard';
+import { getWallets, SUI_CHAINS, ReadonlyWalletAccount } from '@mysten/wallet-standard';
+import { getFaucetHost, requestSuiFromFaucetV0 } from '@mysten/sui.js/faucet';
+import { useEffect } from 'react';
+
+const WALLET_NAME = 'Unsafe Burner Wallet';
+
+export function useUnsafeBurnerWallet(enabled: boolean) {
+	useEffect(() => {
+		if (!enabled) {
+			return;
+		}
+		const unregister = registerUnsafeBurnerWallet();
+		return unregister;
+	}, [enabled]);
+}
+
+function registerUnsafeBurnerWallet(
+	client: SuiClient = new SuiClient({
```

Comment:
I wasn't sure if we should assume `localnet` or try to leverage the client from `SuiClientProvider` 🤔. I suppose the former makes sense for simplicity for now?

Reply:
I think it might be useful to use the provided client, incase you need special connection params (eg custom headers or whatever for a custom node)

Reply:
Yeah, fair point! I switched this to use the provided client